lib/bots/versionbot.ts

```diff
@@ -390,6 +409,228 @@ export class VersionBot extends ProcBot {
+        let approvedReviewers: string[];
+
+        // Only when opening a new PR.
+        console.log(event.cookedEvent.data.action);
+        if (event.cookedEvent.data.action !== 'opened') {
+            return Promise.resolve();
+        }
+
+        this.logger.log(LogLevel.INFO, `Checking reviewers list for ${owner}/${repo}#${pr.number}`);
+
+        // Get the reviewers for the PR.
+        return this.retrieveConfiguration({
+            emitter: this.githubEmitter,
+            location: {
+                owner,
+                repo,
+                path: RepositoryFilePath
+            }
+        }).then((config: VersionBotConfiguration) => {
+            approvedMaintainers = this.stripPRAuthor((config || {}).maintainers || null, pr) || [];
+            approvedReviewers = this.stripPRAuthor((config || {}).reviewers || null, pr) || [];
+
+            return this.dispatchToEmitter(this.githubEmitterName, {
+                data: {
+                    number: pr.number,
+                    owner,
+                    repo
+                },
+                method: this.githubApi.pullRequests.get
+            });
+        }).then((pullRequest: GithubApiTypes.PullRequest) => {
+            // Look at the reviewers, create a list of configured reviewers that were not added
+            // when the PR was opened.
+            const assignedReviewers = _.map(pullRequest.requested_reviewers, (reviewer) => reviewer.login);
+            const configuredReviewers = _.uniq(_.unionWith(approvedReviewers, approvedMaintainers));
+            let missingReviewers = _.filter(configuredReviewers, (reviewer) => {
+                return !(_.find(assignedReviewers, (assignedReviewer) => (assignedReviewer === reviewer)) ||
+                    (reviewer === pr.user.login));
+            });
+
+            // We don't assign the author as a reviewer, if they're in the list.
+            // An App has no ability to create a review request. The best we can do is to create a new
+            // comment pinging those missing in the list to add themselves.
+            if (missingReviewers.length > 0) {
+                let reviewerMessage = '';
+                _.each(missingReviewers, (reviewer) => {
+                    reviewerMessage += `@${reviewer}, `;
+                });
+                reviewerMessage += ReviewerAddMessage;
+
+                return this.dispatchToEmitter(this.githubEmitterName, {
+                    data: {
+                        owner,
+                        repo,
+                        number: pr.number,
+                        body: reviewerMessage
+                    },
+                    method: this.githubApi.issues.createComment
+                });
+            }
+        });
+    }
+
+    /**
+     * Checks to ensure that the minimum number of approvals for a PR occurs.
+     * Should the conditions be satisfied then a successful status is set, otherwise a failed
+     * state is set.
```

Comment:
No, this is a new status check, and as such it's registered in the constructor. It doesn't get called from any of the other methods. See the `ghListener.registerEvent` call in the constructor, it's the second registration object.